server/game/cards/03_TWI/upgrades/AhsokasPadawanLightsaber.ts

```diff
@@ -0,0 +1,30 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { Card } from '../../../core/card/Card';
+import { Trait } from '../../../core/Constants';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+
+export default class AhsokasPadawanLightsaber extends UpgradeCard {
+    protected override getImplementationId() {
+        return {
+            id: '0741296536',
+            internalName: 'ahsokas-padawan-lightsaber'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setAttachCondition((card: Card) => !card.hasSomeTrait(Trait.Vehicle));
+
+        this.addWhenPlayedAbility({
+            title: 'Attack with Ahsoka Tano',
```

Comment:
This implementation is wrong, the card ability is to attack with any unit not necessarily Ahsoka

Reply:
Alright, this should be fixed up and test passing. The one question I had: Should the `optional` be on the When Played or should it be on the attack? The attack is technically optional, but seems the user would expect to pass the whole When Played trigger instead of selecting a card then passing on the attack.